src/components/OrdersWidget/OrderRow.tsx

```diff
@@ -130,16 +131,27 @@ const Amounts: React.FC<AmountsProps> = ({ sellToken, order }) => {
 
   let filledCellContent = 'no limit'
   let totalCellContent = 'no limit'
+  let amountFilled: string | null = null
 
   if (!order.isUnlimited) {
     filledCellContent = `${filledAmount} ${displayTokenSymbolOrLink(sellToken)}`
     totalCellContent = `${totalAmount} ${displayTokenSymbolOrLink(sellToken)}`
+
+    // Calculate filled amount percentage
+    // need to convert BN values to Bignumber for decimals math...
+    const filledAmt = parseBigNumber(filledAmountBN.toString())
+    const totalAmt = parseBigNumber(order.priceDenominator.toString())
+    const calculatable = filledAmt && totalAmt && !filledAmt.isZero()
+    // TS hates the above conditional statements, filledAmt and totalAmt are not NULL when calculatable passed ternary!
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    amountFilled = calculatable ? filledAmt!.div(totalAmt!).times('100').toFixed(2) : null
```

Comment:
You can specify the whole shebang instead of `calculatable` then it'll recognize as not null
Or do assignment inside `if (filledAmt && totalAmt && !filledAmt.isZero())`

Reply:
yeah that's what I normally do but @anxolin has suggested compacting into vars (which i agree is more readable) but i think i stopped doing so subconsciously because of this very reason with TS. Anyways thought I'd comment to see if you had an answer for me and you did! ;)